### PR TITLE
Javadocs version 1 complete

### PR DESCRIPTION
**Pull request by albydeca**
Hi, I have finished creating the first version of the Java SDK documentation on the Wiki. As suggested this closely follows the NodeJS docs.

One issue I would flag is that I am unable to see syntax highlighting for the many Java snippets I have included, despite this highlighting working perfectly in VSCode. Any ideas why this is?

Please let me know what you think. I'm tagging @mastrogiovanni as he is my goto developer at the Foundation.

Cheers :)

### CODE REVIEW

**Comment by mastrogiovanni** [comment]
Hi @albydeca can you update the documentation so that conflicting files are solved?

File: documentation/docs/basics/authentication/example_java.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+					statusCode + EntityUtils.toString(response_body, StandardCharsets.UTF_8));
+		}
+		return response_body;
+	}
+
+```
+
+##  Putting It All Together
+
+You can find the complete code example [here](https://github.com/albydeca/iota-is-sdk/blob/main/src/main/java/net/gradbase/clients/BaseClient.java). All snippets above are taken from there.
```
Comment (aleksei-korolev) [inline documentation/docs/basics/authentication/example_java.md:210]: @mastrogiovanni where should we store the example source code?

File: documentation/docs/examples/java/create-channel.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+keywords:
+- create channel
+- read channel
+- write channel
+- example
+---
+# Create Channel
+
+The [example-5](https://github.com/albydeca/iota-is-sdk/blob/main/examples/src/main/java/net/gradbase/examples/CreateChannel.java)
+script authenticates an Integration Service client to manage Identities using the Admin identity created in [example-0](how-to-run-examples) and then performs the following tasks:
```
Comment (aleksei-korolev) [inline documentation/docs/examples/java/create-channel.md:13]:
```suggestion
script authenticates an Integration Services client to manage identities using the admin identity created in [example-0](how-to-run-examples) and then performs the following tasks:
```

identity and admin identity: these are just technical terms, no need to capitalize technical terms. An Integration Services client: a client is a technical term, so non-capitalized, Integration Services is a name, so capitalized. Same across the rest of the contribution. No worries, this is a confusing rule.

File: documentation/docs/examples/java/create-identity-and-credentials.mdx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+```bash
+TODO
+```
+
+## Authentication
+
+You can use the following script to authenticate the client using the Admin identity and get the
+BasicIdentityCredential that the admin owns: this VC (verifiable credential),
+which is required for an Issuer to sign verifiable credentials for other Identities.
```
Comment (aleksei-korolev) [inline documentation/docs/examples/java/create-identity-and-credentials.mdx:32]:
```suggestion
You can use the following script to authenticate the client using the admin identity and get the
BasicIdentityCredential that the admin owns: this is a verifiable credential
which is required for an issuer to sign verifiable credentials for other Identities.
```

File: documentation/docs/examples/java/delete-users.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+You can remove an identity from the Integration Services backend by running the following script:
+
+```java
+// Remove the user and also revoke the user's credentials
+identityClient.remove(someIdentity.getId(), true);
+```
+
+:::note
+You can only remove an identity in the Integration Services backend, **the Identity cannot be removed from the Tangle**.
```
Comment (aleksei-korolev) [inline documentation/docs/examples/java/delete-users.md:39]:
```suggestion
You cannot delete an identity from the Tangle. You can only remove it from the Integration Services database.
```

You can emphasise important information by simply putting it first.

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+The Java SDK assumes the existence of a running instance of the Integration Services API, the endpoint of which is to be specified in the `api-url` part of the config file.
+
+An `adminIdentity.json` is also strongly recommended to make the best use of the codebase. This file should contain the details of a root identity with CRUD capabilities.
+
+The codebase is divided into three parts:
+
+* `IdentityClient` allows the developer to create, update, delete users, add/revoke/check their credentials and manage trusted authorities. Self-Sovereign Identity (SSI) developers should make use of this part.
+* `ChannelClient` allows the developer to create, delete data channels (IOTA Streams), manage all subscription-related actions for those channels, push and pull data. IoT developers might want to use this to implement M2M communications.
+* `examples` provides a set of sample activities that show the potential of what can be done with the codebase, and also acts as a rudimentary testsuite. More about these can be found in the [dedicated section](../../../examples/java/introduction.mdx).
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:91]:
```suggestion
* `examples` provides a set of sample activities that show the potential of what can be done with the codebase, and also acts as a rudimentary test suite. More about these can be found in the [dedicated section](../../../examples/java/introduction.mdx).
```

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+Then, perform the following steps:
+```bash
+mvn clean install
+```
+You are now ready to use the JAR and access the classes. Please remember to keep the `env.properties` *in the same folder as the JAR*.
+The JAR can be used as a dependency to run the examples, which, contrary to the node implementation, exist as part of their own package
+which depends on this project's JAR (see `examples/pom.xml` in the repo)
+
+Note that the sensitive variables are accessed in the code as follows:
+```java
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:69]:
```suggestion
Note that the sensitive variables are accessed in the code as follows:

```java
```

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  api-version=vX.X
+  api-url=XXXXXXX
+  identity-file=adminIdentity.json
+- `adminIdentity.json` - will contain the admin identity object (json file with elements `doc` and `key`)
+
+Then, perform the following steps:
+```bash
+mvn clean install
+```
+You are now ready to use the JAR and access the classes. Please remember to keep the `env.properties` *in the same folder as the JAR*.
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:64]:
```suggestion

You are now ready to use the JAR and access the classes. Please remember to keep the `env.properties` *in the same folder as the JAR*.
```

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+### Run Integration Services API
+
+The Java SDK assumes the existence of a running instance of the Integration Services API, the endpoint of which is to be specified in the `api-url` part of the config file.
+
+An `adminIdentity.json` is also strongly recommended to make the best use of the codebase. This file should contain the details of a root identity with CRUD capabilities.
+
+The codebase is divided into three parts:
+
+* `IdentityClient` allows the developer to create, update, delete users, add/revoke/check their credentials and manage trusted authorities. Self-Sovereign Identity (SSI) developers should make use of this part.
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:89]:
```suggestion
* `IdentityClient` allows the developer to create, update, delete users and add, revoke, and check their credentials, and manage trusted authorities. Self-Sovereign Identity (SSI) developers should make use of this part.
```

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+public BaseClient() throws FileNotFoundException, IOException {
+    Properties appProps = new Properties();
+    appProps.load(new FileInputStream("env.properties"));
+    this.apiVersion = appProps.getProperty("api-version");
+    this.apiKey = appProps.getProperty("api-key");
+    this.baseUrl = appProps.getProperty("api-url") + apiVersion + "/";
+    this.jwt = null;
+}
+```
+Anyone who wishes to implement such access in a more convenient way for their needs is welcome to make the changes and submit a pull request.
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:79]:
```suggestion

Anyone who wishes to implement such access in a more convenient way for their needs is welcome to make the changes and submit a pull request.
```

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+  api-key=XXXXXXX
+  api-version=vX.X
+  api-url=XXXXXXX
+  identity-file=adminIdentity.json
+- `adminIdentity.json` - will contain the admin identity object (json file with elements `doc` and `key`)
+
+Then, perform the following steps:
+```bash
+mvn clean install
+```
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:63]: I think you are missing triple ticks here somewhere.

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+Build from source
+```bash
+git clone git@github.com:albydeca/iota-is-sdk.git
+cd iota-is-sdk
+mvn clean install
+```
+
+or download JAR from [MVNRepository](https://mvnrepository.com/artifact/net.gradbase/iota.is.sdk/0.0.1)
+
+or simply add to your POM:
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:38]:
```suggestion
or simply add to your POM:

```

File: documentation/docs/getting_started/installation/java/local_setup.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+See project's [POM](https://github.com/albydeca/iota-is-sdk/blob/main/pom.xml). Moreover, the following will be required:
+
+* JDK 16 (recommended)
+* A reference to an instance of the [Integration Services API](https://github.com/iotaledger/integration-services)
+* Maven
+
+## Download and Build the Project
+
+Build from source
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started/installation/java/local_setup.md:29]:
```suggestion
Build from source:

```

File: documentation/docs/examples/java/introduction.mdx [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+  }
+}
+```
+
+If you have a JSON Identity, you can authorize your client as follows:
+
+```java
+  client.authenticate(newUserIdentity.getJSONObject("doc").getString("id"),
+			  newUserIdentity.getJSONObject("key").getString("public"),
+				newUserIdentity.getJSONObject("key").getString("secret"));
```
Comment (aleksei-korolev) [inline documentation/docs/examples/java/introduction.mdx:135]: Would it make sense to introduce local variables to make the example more readable?

File: documentation/docs/examples/java/introduction.mdx [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+  IdentityClient client = new IdentityClient();
+  JSONObject jsonClaim = new JSONObject().put("type", "Person").put("name", "randomName");
+
+  Claim claim = new Claim(jsonClaim);
+
+  final String randomUsername = Utils.getRandomUsernameOfLength(5);
+  System.out.println("username: " + randomUsername);
+  JSONObject newUserIdentity = client.create(randomUsername, claim);
+  System.out.println("created new user " + newUserIdentity);
+
```
Comment (aleksei-korolev) [inline documentation/docs/examples/java/introduction.mdx:95]:
```suggestion
```

File: documentation/docs/examples/java/introduction.mdx [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+The following sections  will guide you in how to use the [Integration Services SDK](https://www.npmjs.com/package/iota-is-sdk).
+The Integration Services SDK makes it easy to manage decentralized identities and secure channels (Audit Trail).
+
+You should have a basic understanding of [decentralized identities](../../basics/identity)
+to get the most out of the following examples.
+
+## Install
+
+Build from source
```
Comment (aleksei-korolev) [inline documentation/docs/examples/java/introduction.mdx:23]:
```suggestion
Build from source:

```

**Review by aleksei-korolev — CHANGES_REQUESTED** [review]
Hi @albydeca, I am one of the technical writers at IF. Thank you for your contribution, I have just finished reviewing it. Please don't be discouraged by the amount of change requests, these are mostly minor formatting and styling fixes. All in all, your contribution is solid.

Cheers

File: documentation/docs/basics/authentication/example_java.md [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+270c2e502c5c753e39159683981e452444f81a10d798f56406a9c471d672a5ede1792cb7f97d4f9c9efeec7bf35577dd1f8482afca7e3710291868a65bf91e07
+```
+
+### 4. Request the JWT
+
+The last step is to request the JWT. You can use the following code snippet with your signed nonce in the request body.
+
+```java
+    [...]
+        final String endpoint = "authentication/prove-ownership/" + didId;
```
[the remark this answers is not in the document]
Comment (albydeca) [reply documentation/docs/basics/authentication/example_java.md:144]: yes. Sorry about that

File: documentation/docs/getting_started/installation/java/local_setup.md (hunk 10 above)
Comment (albydeca) [reply documentation/docs/getting_started/installation/java/local_setup.md]: indeed :)

**Comment by albydeca** [comment]
Hi all,

Thanks to @aleksei-korolev for the useful suggestions! I have now implemented the changes.

As for the conflicts, most of them stem from the fact that I have changed the directory structure in the `examples/` folder, in particular I created two subfolders: `node/` and `java/`. The rest of the conflicts are in files with minimum intervention on my part. It may be that my changes are conflicting with newer ones on the master branch.

It might make sense for us to get on a call to better understand how we can deal with this. Apart from the changes in directory structure, the rest should be immediate to solve.

**Review by mastrogiovanni — APPROVED** [review]

**Pull request merged**